### PR TITLE
Update to Firmware 7.25

### PR DESCRIPTION
**Pull request by SmokingDevices**
Update to latest firmware version.

### CODE REVIEW

File: src/main/java/com/github/kaklakariada/fritzbox/model/homeautomation/ColorControl.java [hunk 1]
```diff
@@ -38,6 +38,18 @@
...
     private String current_mode;
 
+    @Attribute(name = "fullcolorsupport", required = false)
+    private String fullcolorsupport;
+
+    @Attribute(name = "mapped", required = false)
+    private String mapped;
+
+    @Element(name = "unmapped_hue", required = false)
+    private String unmapped_hue;
```
Comment (kaklakariada) [inline src/main/java/com/github/kaklakariada/fritzbox/model/homeautomation/ColorControl.java]: Please rename the field to use camelCase. Same for the other attributes and getters.

File: src/main/java/com/github/kaklakariada/fritzbox/model/User.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.github.kaklakariada.fritzbox.model;
+
+import org.simpleframework.xml.Attribute;
+
+public class User {
```
Comment (kaklakariada) [inline src/main/java/com/github/kaklakariada/fritzbox/model/User.java:23]: According to the xml, the user also has a name:

```xml
    <Users>
        <User>UserA</User>
        <User>UserB</User>
        <User last="1">UserC</User>
    </Users>
```
Please add a field for the name.

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Thanks for your pull request! Please fix my findings, then we can merge this.

File: src/main/java/com/github/kaklakariada/fritzbox/model/User.java (hunk 2 above)
Comment (SmokingDevices) [reply src/main/java/com/github/kaklakariada/fritzbox/model/User.java:23]: implemented

File: src/main/java/com/github/kaklakariada/fritzbox/model/homeautomation/ColorControl.java (hunk 1 above)
Comment (SmokingDevices) [reply src/main/java/com/github/kaklakariada/fritzbox/model/homeautomation/ColorControl.java]: done

File: src/main/java/com/github/kaklakariada/fritzbox/model/User.java (hunk 2 above)
Comment (SmokingDevices) [reply src/main/java/com/github/kaklakariada/fritzbox/model/User.java:23]: implemented

File: src/main/java/com/github/kaklakariada/fritzbox/model/User.java [hunk 3]
```diff
@@ -18,13 +18,21 @@
...
 public class User {
 
     @Attribute(name = "last", required = false)
-    public boolean last;
+    private int last;
+
+    @Text
+    private String name;
 
     public boolean isLast() {
```
Comment (kaklakariada) [inline src/main/java/com/github/kaklakariada/fritzbox/model/User.java:31]: Sorry for nit-picking. But if this returns an int, we should rename the getter.
```suggestion
    public boolean getLast() {
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Comment by kaklakariada** [comment]
Please also add an entry to the Changelog. Previously i always did this myself, but you know better how to describe your changes :)

File: src/main/java/com/github/kaklakariada/fritzbox/model/User.java (hunk 3 above)
Comment (SmokingDevices) [reply src/main/java/com/github/kaklakariada/fritzbox/model/User.java:31]: According to the [documentation](https://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/AVM_Technical_Note_-_Session_ID_deutsch_Dez2020.pdf) (Page 2: "Bedeutung der Werte:") this should indicate the last user which then (from my point of view) is a boolean.

My poor XML skills didn't bring me to a solution where i could mark the field to be a boolean and the xml annotation would do the translation from 1 to true.

File: src/test/java/com/github/kaklakariada/fritzbox/mapping/DeserializerTest.java [hunk 4]
```diff
@@ -78,4 +80,25 @@ public void parseDeviceListAllTogether() throws IOException {
...
 
+    @Test
+    public void parseDeviceList() throws IOException {
+        final String fileContent = Files.readAllLines(Paths.get("src/test/resources/deviceList.xml")).stream()
+                .collect(joining("\n"));
+        new Deserializer().parse(fileContent, DeviceList.class);
+    }
+
+    @Test
+    public void parseSessionInfo() throws IOException {
```
Comment (PeterJelitsch) [inline src/test/java/com/github/kaklakariada/fritzbox/mapping/DeserializerTest.java:106]: [Minor] With my from today PR I just introduced assertJ into the project. Maybe you want to have a look at those Assertion classes which allow fluent-API assertions. BR, Peter.

Comment (SmokingDevices) [reply src/test/java/com/github/kaklakariada/fritzbox/mapping/DeserializerTest.java:106]: Will have a look, but as rebase is not my favourite git command, i would update these tests with my next contribution.

Comment (kaklakariada) [reply src/test/java/com/github/kaklakariada/fritzbox/mapping/DeserializerTest.java:106]: Rebasing ist not required. I also don't use it ;)

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,7 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [1.4.0] - unreleased
 
 ### Added
-
+* [#18] Added new fields to be compatible with Fritz!OS 7.25
```
Comment (kaklakariada) [inline CHANGELOG.md:10]: 👍

File: src/main/java/com/github/kaklakariada/fritzbox/model/User.java (hunk 3 above)
Comment (kaklakariada) [reply src/main/java/com/github/kaklakariada/fritzbox/model/User.java:31]: 👍

**Review by kaklakariada — APPROVED** [review]

**Comment by kaklakariada** [comment]
I just created a new release 1.4.0 and published it to Maven Central. It may take a while until it is available at https://repo1.maven.org/maven2/com/github/kaklakariada/fritzbox-java-api/

**Pull request merged**